File: frontend/city-market/src/app/services/stock/istock.service.ts
import {Stock} from '../../entities/stock';
import {Market} from '../../entities/merket';

/**
 * Interface for working with product storage.
 */
export interface IStockService {
  /**
   * The method saves the created store.
   *
   * @param {Stock} stock
   */
  saveStock(stock: Stock);

  /**
   *  The method creates a query in the database to obtain a list of warehouses.
   */
  getStocksToPromise();

  /**
   * The method returns the stock registered for the actions.
   */
  getSelectedStock();

  /**
   * The method establishes a warehouse for the work to be carried out.
   *
   * @param {Stock} stock
   */
  setSelectedStock(stock: Stock);

  /**
   * The method forms a query in the database to obtain all the warehouses to which the market is
   * subscribed.
   *
   * @param {Market} market
   */
  getStocksByMarket(market: Market);

  /**
   * The method signs the store on the storage list.
   *
   * @param {Market} market
   * @param {Stock[]} stock
   */
  signStockMarket(market: Market, stock: Stock[]);

}

File: src/main/java/by/intexsoft/service/IStockService.java
package by.intexsoft.service;

import java.util.List;
import by.intexsoft.entity.Market;
import by.intexsoft.entity.Stock;
import by.intexsoft.entity.User;
import by.intexsoft.repository.MarketRepository;
import by.intexsoft.repository.UserRepository;

/**
 * Service for working with the repository, which processes information about
 * warehouses. Using this class, you can receive, store, modify information
 * about the repositories.
 * 
 * @see {@link User}, {@link UserRepository}
 */
public interface IStockService {
	/**
	 * Gets the stock from the repository by its name.
	 * 
	 * @param nameStock
	 *            store name.
	 * @return {@link Stock}
	 */
	public Stock findByName(String nameStock);

	/**
	 * Saving a new storage.
	 * 
	 * @param stock
	 *            instance @link {@link Stock}
	 */
	public Stock save(Stock stock);

	/**
	 * Returns all stocks from the repository.
	 * 
	 * @return {@link List} <{@link Stock}>
	 */
	public List<Stock> findAll();

	/**
	 * Returns the store by its id.
	 */
	public Stock findOne(int id);

	/**
	 * Deletes the stock by id.
	 */
	public void delete(int id);

	/**
	 * The service method returns the list of warehouses for which the warehouse is
	 * subscribed.
	 * 
	 * @see {@link Stock} , {@link MarketRepository}
	 */
	public List<Stock> finfByMasrket(Market market);

	/**
	 * A service method that removes all links from the database and creates new
	 * ones that are passed by the list in the input parameters.
	 * 
	 * @see {@link Stock}
	 */
	public List<Stock> signStockforMarket(List<Stock> stocks, int idMarket);
}
